QEfficient/transformers/models/modeling_auto.py

```diff
@@ -367,30 +353,277 @@ def generate(
+
+        # Initialize the model using from_pretrained similar to transformers.AutoModel.
+        model = QEFFAutoModel.from_pretrained("model_name")
+
+        # Now you can directly compile the model for Cloud AI 100
+        model.compile(num_cores=16, device_group=[0])  # Considering you have a Cloud AI 100 SKU
+
+        #prepare input
+        tokenizer = AutoTokenizer.from_pretrained(model_name)
+        inputs = tokenizer("My name is", return_tensors="pt")
+
+        # You can now execute the model
+        model.generate(inputs)
+    """
+
     _hf_auto_class = AutoModel
-    _pytorch_transforms = [AwqToMatmulNbitsTransform, GPTQToMatmulNbitsTransform, CustomOpsTransform]
-    _onnx_transforms = [FP16ClipTransform, SplitTensorsTransform]
+    _pytorch_transforms = [CustomOpsTransform, AwqToMatmulNbitsTransform, GPTQToMatmulNbitsTransform]
+    _onnx_transforms = [FP16ClipTransform]
 
-    def export(self):
-        raise NotImplementedError("Reached too far!!")
+    def __init__(self, model: nn.Module, **kwargs):
+        super().__init__(model)
+        self.model.config.use_cache = True
+        self.num_layers = model.config.num_hidden_layers
+        self.tokenizer = load_hf_tokenizer(self.model.config.name_or_path)
+
+    @classmethod
+    @with_replaced_quantizers
+    def from_pretrained(cls, pretrained_model_name_or_path, *args, **kwargs):
+        """
+        This method serves as the easiest entry point into using QEfficient. The interface is designed to be similar to transformers.AutoModel.
+        Once the model is initialized, you can use other methods such as export, compile, and generate on the same object.
+
+        Args:
+            :pretrained_name_or_path (str): Model card name from HuggingFace or local path to model directory.
+            :args, kwargs: Additional arguments to pass to transformers.AutoModel.
+
+        .. code-block:: python
+
+            from QEfficient import QEFFAutoModel
+            from transformers import AutoTokenizer
+
+            # Initialize the model using from_pretrained similar to transformers.AutoModel.
+            model = QEFFAutoModel.from_pretrained("model_name")
+
+            # Now you can directly compile the model for Cloud AI 100
+            model.compile(num_cores=16, device_group=[0])  # Considering you have a Cloud AI 100 SKU
+
+            #prepare input
+            tokenizer = AutoTokenizer.from_pretrained(model_name)
+            inputs = tokenizer("My name is", return_tensors="pt")
+
+            # You can now execute the model
+            model.generate(inputs)
+        """
+        if kwargs.get("attn_implementation", None) not in {None, "eager"}:
+            logger.warning('Updating attn_implementation="eager"')
+
+        if kwargs.get("low_cpu_mem_usage", None):
+            logger.warning("Updating low_cpu_mem_usage=False")
+
+        kwargs.update({"attn_implementation": "eager", "low_cpu_mem_usage": False, "add_pooling_layer": False})
+        try:
```

Comment:
add a comment explaining why we need try/except block and also raise warning that we are removing pooling layer if it exists.